File: src/main/kotlin/com/natu/ftax/IDgenerator/infrastructure/IdGeneratorConfig.java
package com.natu.ftax.IDgenerator.infrastructure;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.time.Clock;

@Configuration
public class IdGeneratorConfig {

    @Bean
    public Clock clock() {
        return Clock.systemUTC();
    }

    @Bean
    public SnowflakeIDGenerator idGenerator(@Value("${instance.id}") int instanceId, Clock clock) {
        return new SnowflakeIDGenerator(instanceId, clock);
    }
}

File: src/main/kotlin/com/natu/ftax/IDgenerator/infrastructure/IdGeneratorConfig.kt
package com.natu.ftax.IDgenerator.infrastructure

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.time.Clock

@Configuration
class IdGeneratorConfig {

    @Bean
    fun clock(): Clock {
        return Clock.systemUTC()
    }

    @Bean
    fun idGenerator(@Value("\${instance.id}") instanceId: Int, clock: Clock): SnowflakeIDGenerator =
        SnowflakeIDGenerator(instanceId, clock)

}